RSpec changes needed for Neo4j Orm Adapter

andreasronge opened the pull request:
Some RSpec changes (bug fixes) in order to make the neo4j orm_adapter happy.
- Using :offset and :limit does not make sense without using :order in `find_all`
- RSpec 3.0 was released. Fixed one spec so it works both for RSpec 2.x and 3.x

andreasronge committed:
Use =~ RSpec match when we can't assume the order returned from find_all and made it work with new final RSpec 3.0

andreasronge committed:
Fixed RSpecs - Using :offset and :limit does not make sense without using :order

josevalim commented on spec/orm_adapter/example_app_shared.rb:
```diff
@@ -134,14 +136,14 @@ def reload_model(model)
           user2 = create_model(user_class, :name => "Fred")
           user3 = create_model(user_class, :name => "Betty")
-          user_adapter.find_all(:name => "Fred").should == [user1, user2]
+          user_adapter.find_all(:name => "Fred").to_a.should =~ [user1, user2]
```
Hello @andreasronge ! How is it going?

Question: do you think .to_a makes sense here? Shouldn't neo4j adapter return an array? Or, to change slightly the question, shouldn't neo4j adapter return the same as the others?

andreasronge replied:
Hi there
Well it does not work for ActiveRecord either and I think we should not assume `find_all` to return an array. (Neo4j takes all my spare time :) so still no time for playing around with elixir :(  )

Without #to_a

```
  1) [ActiveRecord orm adapter] it should behave like example app with orm_adapter adapter instance #find_all (conditions) should return only models matching conditions
     Failure/Error: user_adapter.find_all(:name => "Fred").should =~ [user1, user2]
       expected: [#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>, #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>]
            got: [#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>, #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>] (using =~)
       Diff:
       @@ -1,3 +1,2 @@
       -[#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>,
       - #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>]
       +[#<ArOrmSpec::User id: 15, name: "Fred", rating: nil>, #<ArOrmSpec::User id: 16, name: "Fred", rating: nil>]

     Shared Example Group: "example app with orm_adapter" called from ./spec/orm_adapter/adapters/active_record_spec.rb:36
     # ./spec/orm_adapter/example_app_shared.rb:139:in `block (5 levels) in <top (required)>'

```

josevalim commented:
It makes sense then, thanks!

For some reason this pull request breaks the build (and the last commit passed just fine). Can you please help me find out what went wrong?

andreasronge commented:
Looks like it is because it now uses RSpec 3.0. Maybe we should lock Gemfile to RSpec 2.x for now.

andreasronge committed:
Lock rspec gem to 2.x

Notice, RSpec 3.x was released a few days ago which made lots of specs to fail because the version was not locked.
E.g. `its` is not in rspec-core any longer.

andreasronge committed:
Found some more Gemfile to lock

josevalim commented:
Ah, there is no Gemfile lock... we should probably add some locks to the repo.

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

ianwhite commented:
Thanks Andreas & José!

btw, The reason there was no Gemfile.lock, was so that we catch errors on travis with newer versions of orms.  Is that a good or a bad idea do you think?

On 6 Jun 2014, at 20:01, José Valim notifications@github.com wrote:

> —
> Reply to this email directly or view it on GitHub.

josevalim commented:
Oh, that's a good point. The downside though is pull requests can fail for unrelated reasons. So I am not sure on which side to err.

andreasronge commented:
I think it is good. Now we know that it does not work with latest released RSpec :)

ianwhite commented:
Ok, I'll do some work making sure everything works with the latest and greatest.

On 6 Jun 2014, at 22:56, Andreas Ronge notifications@github.com wrote:

> I think it is good. Now we know that it does not work with latest released RSpec :)
> 
> —
> Reply to this email directly or view it on GitHub.
